consensus/spos/consensusMessageValidator.go

```diff
@@ -290,3 +304,33 @@ func (wrk *Worker) checkMessageWithFinalInfoValidity(cnsMsg *consensus.Message)
 
 	return nil
 }
+
+func (wrk *Worker) isMessageTypeLimitReached(pk []byte, msgType consensus.MessageType) bool {
+	wrk.mutPkConsensusMessages.RLock()
+	defer wrk.mutPkConsensusMessages.RUnlock()
+
+	mapMsgType, ok := wrk.mapPkConsensusMessages[string(pk)]
```

Comment:
Maybe we should have the key as a concatenation between pk and round. This will take care of the edgecases in which a node receives 2 messages of the same type but in consecutive rounds caused by the small differences between synced clocks.

Reply:
We can do this just to be super safe, but if the node receives a consensus message with the previous round inside, it will drop this message in checkConsensusMessageValidity method in line 66 so it will not be added in this map